vdev/src/testing/config.rs

```diff
@@ -37,20 +37,49 @@ impl RustToolchainConfig {
     }
 }
 
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, Serialize)]
 pub struct ComposeConfig {
     pub services: BTreeMap<String, ComposeService>,
-    #[serde(default)]
+    #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
     pub volumes: BTreeMap<String, Value>,
+    #[serde(default)]
+    pub networks: BTreeMap<String, BTreeMap<String, String>>,
 }
 
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, Serialize)]
 pub struct ComposeService {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub image: Option<String>,
```

Comment:
I was under the assumption that `image` was required, in what case can it be `None`?

Reply:
This might have been an artifact of me being copy pasta happy with that line above and putting it over even fields that are required or guaranteed to be included.

Reply:
Ah but in this case it was added because `cargo vdev integration start dnstap` 

`scripts/integration/dnstap/compose.yaml`
```yaml
version: "3"

services:
  dnstap:
    build:
      context: ../../../tests/data/dnstap
    container_name: vector_dnstap
    hostname: ns.example.com
    volumes:
      - dnstap-sockets:/bind1/etc/bind/socket
      - dnstap-sockets:/bind2/etc/bind/socket
      - dnstap-sockets:/bind3/etc/bind/socket

volumes:
  dnstap-sockets: {}
```

Reply:
noting that `image` isn't required when defining the test runner service, which builds from the local source.

Reply:
But it is required to capture and recreate the `image` parameter already existing in the compose files.

Reply:
ah, yeah I was just adding a case example to Spencer's earlier question, could have quoted that

> I was under the assumption that image was required, in what case can it be None?